frontend/src/tabs-hook.tsx

```diff
@@ -31,82 +39,60 @@ class TabsHook extends Logger {
-          }
-          if (qAMRoot?.child && !qAMRoot?.child?.type?.decky) {
-            afterPatch(qAMRoot.child, 'type', (_: any, ret: any) => {
-              try {
-                const qamTabsRenderer = findInReactTree(ret, (x) => x?.props?.onFocusNavDeactivated);
-                if (patchedInnerQAM) {
-                  qamTabsRenderer.type = patchedInnerQAM;
-                } else {
-                  afterPatch(qamTabsRenderer, 'type', (innerArgs: any, ret: any) => {
-                    const tabs = findInReactTree(ret, (x) => x?.props?.tabs);
-                    this.render(tabs.props.tabs, innerArgs[0].visible);
-                    return ret;
-                  });
-                  patchedInnerQAM = qamTabsRenderer.type;
-                }
-              } catch (e) {
-                this.error('Error patching QAM inner', e);
+  async init() {
+    this.qAMRoot = await this.getQuickMenuRoot();
+
+    let patchedInnerQAM: any;
+    this.qamPatch = afterPatch(this.qAMRoot.return, 'type', (_: any, ret: any) => {
+      try {
+        if (this.qAMRoot?.child && !this.qAMRoot?.child?.type?.decky) {
+          afterPatch(this.qAMRoot.child, 'type', (_: any, ret: any) => {
+            try {
+              const qamTabsRenderer = findInReactTree(ret, (x) => x?.props?.onFocusNavDeactivated);
+              if (patchedInnerQAM) {
+                qamTabsRenderer.type = patchedInnerQAM;
+              } else {
+                afterPatch(qamTabsRenderer, 'type', (innerArgs: any, ret: any) => {
+                  const tabs = findInReactTree(ret, (x) => x?.props?.tabs);
+                  this.render(tabs.props.tabs, innerArgs[0].visible);
+                  return ret;
+                });
+                patchedInnerQAM = qamTabsRenderer.type;
               }
-              return ret;
-            });
-            qAMRoot.child.type.decky = true;
-            qAMRoot.child.alternate.type = qAMRoot.child.type;
-          }
-        } catch (e) {
-          this.error('Error patching QAM', e);
+            } catch (e) {
+              this.error('Error patching QAM inner', e);
+            }
+            return ret;
+          });
+          this.qAMRoot.child.type.decky = true;
+          this.qAMRoot.child.alternate.type = this.qAMRoot.child.type;
         }
+      } catch (e) {
+        this.error('Error patching QAM', e);
+      }
 
-        return ret;
-      });
+      return ret;
+    });
 
-      if (qAMRoot.return.alternate) {
-        qAMRoot.return.alternate.type = qAMRoot.return.type;
-      }
-      this.log('Finished initial injection');
-    })();
+    if (this.qAMRoot.return.alternate) {
+      this.qAMRoot.return.alternate.type = this.qAMRoot.return.type;
+    }
+    this.log('Finished initial injection');
+  }
+
+  async getQuickMenuRoot() {
+    while (typeof GamepadNavTree === 'undefined') {
+      await sleep(50);
+    }
+
+    const parentNode = findSP().document.querySelector("[class*='BasicUI']");
```

Comment:
Don't hardcode a specific class this way, use the static classes DFL provides. also, this is a class namespace, not a specific class, which means it can return unintended results